tests/e2e/tests/all-reviews/all-reviews.block_theme.spec.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import { expect, test } from '@woocommerce/e2e-playwright-utils';
+import { default as WooCommerceRestApi } from '@woocommerce/woocommerce-rest-api';
+
+const blockData = {
+	name: 'woocommerce/all-reviews',
+	selectors: {
+		frontend: {
+			firstReview:
+				'.wc-block-review-list-item__item:first-child .wc-block-review-list-item__text p',
+		},
+	},
+};
+
+test.describe( `${ blockData.name } Block`, () => {
+	let productId: number;
+	let firstReviewId: number;
+	let secondReviewId: number;
+
+	// Create product and reviews.
+	test.beforeAll( async ( { baseURL } ) => {
+		const api = new WooCommerceRestApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
```

Comment:
Good questions, I'm not 100% sure, to be honest. I copied this approach [from WC core](https://github.com/woocommerce/woocommerce/blob/f35d52203ea49443113a2609f787981ba2958e5f/plugins/woocommerce/tests/e2e-pw/tests/shopper/cart.spec.js#L9-L49), but I'm not sure whether there is a better way of doing that. Pinging @dinhtungdu as he knows a lot about e2e tests.

Reply:
Understood. I raised this question primarily to explore whether we could apply the same approach to other areas of our codebase, including the tests for Product Collection. Gaining some clarity on this aspect would be extremely helpful.

@sunyatasattva I would really appreciate your input here 🙂

Reply:
We're still searching for a good way to prepare the data for testing. I think using WooCommerce REST API is a good approach we should consider standardizing. For now we're creating test data at the provisioning time, and there is clear needs to create test data/fixture at the test run time. Manish also did some research here for the Seville meetup: per0F9-13G-p2.

> Currently, our tests run serially, but I'm guessing we'll be shifting to parallel testing soon.

Even we aim to run the test in parallel and we should write tests in isolation, I think we should only enable parallelism on local, not CI. According to the Playwright docs:

> We recommend setting [workers](https://playwright.dev/docs/api/class-testconfig#test-config-workers) to "1" in CI environments to prioritize stability and reproducibility. Running tests sequentially ensures each test gets the full system resources, avoiding potential conflicts. However, if you have a powerful self-hosted CI system, you may enable [parallel](https://playwright.dev/docs/test-parallel) tests. For wider parallelization, consider [sharding](https://playwright.dev/docs/test-parallel#shard-tests-between-multiple-machines) - distributing tests across multiple CI jobs.
> https://playwright.dev/docs/ci#workers

Reply:
> In this case: is there a reason why can't we add a product with reviews on our initial fixture to test these use-cases?

AFAIK there was a related discussion #10099, where I proposed a standard approach for adding new fixture data, leading to task #10189. The main reason we haven't done this yet is our limited bandwidth during the cooldown period. Maybe we can prioritize this task to work on it during the cooldown 🤔

Reply:
Thanks for all your feedback, folks! :raised_hands: 

> In this case: is there a reason why can't we add a product with reviews on our initial fixture to test these use-cases?

Not really, it was mostly because I took WC core tests as an example. I guess one advantage of using the API is that we can have different tests using different test data. Ie: if at some point we wanted to test how the Reviews blocks behave when there are no reviews in the store, it would be easier to do so if there are no reviews created during the setup.

In any case, that's not the case right now, as we are only testing these blocks when _there are_ reviews. :slightly_smiling_face: So in cc89fa6f71a004052e37535ba1ea993c78b87bb7 I refactored this PR to create the reviews on setup instead of at the beginning of the test. I also removed `WCRestApiUtils` for now, as it's no longer necessary.

@imanish003 I will need to ask for another review. :pray: